src/rulesets/Base/nondiff.jl

```diff
@@ -88,6 +89,7 @@
 @non_differentiable similar(::AbstractArray{Bool}, ::Any)
 @non_differentiable stride(::AbstractArray{Bool}, ::Any)
 @non_differentiable strides(::AbstractArray{Bool})
+@non_differentiable vcat(::AbstractArray{Bool})
```

Comment:
Should this be `vcat(::AbstractArray{Bool}...)` if that's accepted?

Also, many of these like `strudes` & `isperm` surely aren't differentiable with any input, but `isperm([true])` is an error anyway. 

`similar` also accepts further argumens.

Reply:
Yes, I think that should be the case. We'll wait on the implementation of support of `Vararg`'s as its implemented in PR https://github.com/JuliaDiff/ChainRulesCore.jl/pull/254 ?